crossword.js

```diff
@@ -240,30 +251,54 @@ function reset() {
 }
 
 function editRule(axis, idx) {
-    var rule_span = document.getElementById('rule_' + axis + '_' + idx);
-    rule_span.innerHTML = ruleInput(axis,idx);
+  var rule_span = document.getElementById('rule_' + axis + '_' + idx);
+  rule_span.innerHTML = ruleInput(axis,idx);
 }
 
 function updateRule(axis, idx, value) {
-    board_data[axis][idx] = value;
-    var rule_span = document.getElementById('rule_' + axis + '_' + idx);
-    rule_span.innerHTML = ruleDisplay(axis, idx);
-    checkRules();
+  if (value == '') {
+    value = '.*';
+  }
+  board_data[axis][idx] = value;
+  var rule_span = document.getElementById('rule_' + axis + '_' + idx);
+  rule_span.innerHTML = ruleDisplay(axis, idx);
+  checkRules();
 }
 
 function ruleInput(axis, idx) {
-    var form = '<form action="#" onsubmit="updateRule(\'' + axis + '\', ' + idx + ', rule.value);">'
-    return form + '<input name="rule" value="' + board_data[axis][idx] + '"/></form>';
+  var form = '<form action="#" onsubmit="updateRule(\'' + axis + '\', ' + idx + ', rule.value);">'
+  return form + '<input name="rule" value="' + board_data[axis][idx] + '"/></form>';
 }
 
-
 function ruleDisplay(axis, idx) {
-    var span = '<span ondblclick="editRule(\'' + axis + '\', ' + idx + ')">';
-    return span + board_data[axis][idx] + '</span>';
+  var open = '';
+  var close = '';
+  if (use_editor) {
+    open = '<span ondblclick="editRule(\'' + axis + '\', ' + idx + ');">';
+    close = '</span>';
+  }
+  return open + board_data[axis][idx] + close;
 }
 
 function init() {
   loadData();
+  
+  var url_params = new URLSearchParams(window.location.search);

```

Comment:
`URLSearchParams` is "new" and doesn't work in older browsers, parse window.location directly, or throw this in a try/catch and old browsers just don't get an editor, that's fine =).  Otherwise, looks good!

Reply:
took care of that and niced up the interface a tiny bit